src/connector.rs

```diff
@@ -115,20 +118,82 @@ impl EthConnectorContract {
         let promise1 = sdk::promise_then(
             promise0,
             &account_id,
-            b"finish_deposit",
+            b"finish_deposit_near",
             &data[..],
             NO_DEPOSIT,
             GAS_FOR_FINISH_DEPOSIT,
         );
         sdk::promise_return(promise1);
     }
 
-    pub fn finish_deposit(&mut self) {
+    pub fn deposit_eth(&self) {
+        #[cfg(feature = "log")]
+        sdk::log("[Deposit ETH tokens]".into());
+
+        let proof: Proof = Proof::from(parse_json(&sdk::read_input()).unwrap());
```

Comment:
This is mostly a nit, but I think would improve the code quality.

It looks like a common pattern here to make arguments to these contract methods implicit via `read_input`. I understand that this is needed to get arguments at the runtime level, but I think it would be nice if the core logic did not include these details. For example, in this case we could refactor to have

```rust
/// Contract method
pub fn deposit_eth(&self) {
    let proof: Proof = Proof::from(parse_json(&sdk::read_input()).unwrap());
    let promise = core_deposit_eth(&self, proof);
    sdk::promise_return(promise);
}

/// Core logic for deposit_eth
fn core_deposit_eth(&self, proof: Proof) -> Promise {
    // bulk of code here ...
}
```

I think this is nicer for two reasons:

1. It makes it clear in the code what the method signature is (in this case takes a `Proof` and returns a `Promise`)
2. It makes testing the logic much easier. You can write normal rust unit tests for `core_deposit_eth` and know the smart contract logic is also being tested.

It looks it the refactor suggested above is likely not sufficient to cleanly separate the logic from side-effectful sdk calls, which is really what we want to make unit tests easy to write. But the concept is that even though we need to do some low-level stuff to interface properly with the Near runtime, we should isolate that away from the core logic because it makes things easier to read and test. For example, `Promise` should be a structure that carries all the fields that would be passed to `sdk::promise_create` because then a unit test could check that the output is correct without worrying about mocking up the whole runtime, and all the external methods within.